interop/fake_grpclb/fake_grpclb.go

```diff
@@ -0,0 +1,178 @@
+// The name of the service to load balance for and the addresses
+// of that service are provided by command line flags.
+package main
+
+import (
+	"flag"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	durationpb "github.com/golang/protobuf/ptypes/duration"
+	"google.golang.org/grpc"
+	lbpb "google.golang.org/grpc/balancer/grpclb/grpc_lb_v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/alts"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/testdata"
+)
+
+var (
+	port        = flag.Int("port", 10000, "Port to listen on.")
+	backendIps  = flag.String("backend_addrs", "", "Comma separated list of backend IP/port addresses.")
+	useALTS     = flag.Bool("use_alts", false, "Listen on ALTS credentials.")
+	useTLS      = flag.Bool("use_tls", false, "Listen on TLS credentials, using a test certificate.")
+	shortStream = flag.Bool("short_stream", false, "End the balancer stream immediately after sending the first server list.")
+	serviceName = flag.String("service_name", "UNSET", "Name of the service being load balanced for.")
+)
+
+type backendAddr struct {
+	Ip   []byte
+	Port int32
+}
+
+type loadBalancerServer struct {
+	BackendAddrs []backendAddr
+}
+
+func (l *loadBalancerServer) BalanceLoad(stream lbpb.LoadBalancer_BalanceLoadServer) error {
+	grpclog.Info("Begin handling new BalancerLoad request.")
+	var lbReq *lbpb.LoadBalanceRequest
+	var err error
+	if lbReq, err = stream.Recv(); err != nil {
+		grpclog.Errorf("Error receiving LoadBalanceRequest: %v", err)
+		return err
+	}
+	grpclog.Info("LoadBalancerRequest received.")
+	initialReq := lbReq.GetInitialRequest()
+	if initialReq == nil {
+		grpclog.Info("Expected first request to be an InitialRequest. Got: %v", lbReq)
+		return status.Error(codes.Code(2), "First request not an InitialRequest")
+	}
+	// gRPC clients targetting foo.bar.com:443 can sometimes include the ":443" suffix in
+	// their requested names; handle this case. TODO: make 443 configurable?
+	cleanedName := initialReq.Name
+	if strings.HasSuffix(cleanedName, ":443") {
+		cleanedName = cleanedName[0 : len(cleanedName)-len(":443")]
+	}
+	if cleanedName != *serviceName {
+		grpclog.Info("Expected requested service name: %v. Got: %v", *serviceName, initialReq.Name)
+		return status.Error(codes.Code(2), "Bad requested service name")
+	}
+	if err := stream.Send(&lbpb.LoadBalanceResponse{
+		LoadBalanceResponseType: &lbpb.LoadBalanceResponse_InitialResponse{
+			InitialResponse: &lbpb.InitialLoadBalanceResponse{
+				ClientStatsReportInterval: &durationpb.Duration{
+					Seconds: 1,
+					Nanos:   0,
+				},
+			},
+		},
+	}); err != nil {
+		grpclog.Errorf("Error sending initial LB response: %v", err)
+		return status.Error(codes.Code(2), "Error sending initial response")
+	}
+	serverList := make([]*lbpb.Server, len(l.BackendAddrs))
+	for i := range l.BackendAddrs {
```

Comment:
If we use proto message in the server struct, this won't be necessary.

Reply:
good point, done